### ISSUE TITLE
Search doesn't work when using JSON columns

### ISSUE THREAD

**Issue by RomkaLTU** [yajra/laravel-datatables#1865]
Using DataTables service implementation.

Search doesn't work when selecting from JSON column. Data rendering correctly, but search gives error: `Column not found: 1054 Unknown column 'registries.date' in 'where clause'`

```
$dataTable->columns = [
    ...
    [ 'data' => 'date', 'name' => DB::raw('metas->>"$.date" AS date'), 'title' => 'Data', 'width' => 90 ],
    ...
];

```

- Operating System OSX
- PHP Version 7.2
- Laravel Version 5.5
- Laravel-Datatables Version 8

**Comment by RomkaLTU** [comment]
I was forced to upgrade Laravel from 5.5 to 5.6 and use column definition like this:

`[ 'data' => 'date', 'name' => 'metas->>date', 'title' => 'Data', 'width' => 90 ],`

Some one mentioned that this issue will be fixed in Laravel 5.8 so `metas->date` would be sufficient.

**Comment by oranges13** [comment]
For anyone stuck on Laravel 5.4 who has a similar issue, you can also use `filterColumn` to explicitly tell it how to filter on specific data.

I had to do this to add an implicit relationship filter.

```php
->filterColumn('product.ProductDescription', function($query, $keyword) {
    $query->join(DB::raw('Products'), DB::raw('Products.ProductID'), '=', 'preorders.product_id')->whereRaw('Products.ProductDescription like ?', ["%{$keyword}%"]);
})
```
